core/src/main/resources/reference.conf

```diff
@@ -107,40 +57,101 @@ datastax-java-driver {
   #    session-keyspace = case_insensitive_name
   #    session-keyspace = \"CaseSensitiveName\"
   #
-  # This option is not required; if it is left unspecified, the session won't be connected to any
-  # keyspace, and you'll have to either qualify table names in your queries, or use the per-query
-  # keyspace feature available in Cassandra 4 and above (see Request.getKeyspace()).
+  # If this option is absent, the session won't be connected to any keyspace, and you'll have to
+  # either qualify table names in your queries, or use the per-query keyspace feature available in
+  # Cassandra 4 and above (see Request.getKeyspace()).
   #
   # This can also be provided programatically in CqlSessionBuilder.
-  // session-keyspace = my_keyspace
+  #
+  # Required: no
+  # Modifiable at runtime: no
+  # Overridable in a profile: no
+  // basic.session-keyspace = my_keyspace
 
   # How often the driver tries to reload the configuration.
-  # This option is required, except if you use a different config loader than the driver's built-in
-  # one. Runtime changes are taken into account.
+  #
   # To disable periodic reloading, set this to 0.
-  config-reload-interval = 5 minutes
+  #
+  # Required: yes (unless you pass a different ConfigLoader to the session builder).
+  # Modifiable at runtime: yes, the new value will be used after the next time the configuration
+  #   gets reloaded.
+  # Overridable in a profile: no
+  basic.config-reload-interval = 5 minutes
+
+  basic.request {
```

Comment:
While working on #1010 I noticed that it was hard to figure out the level of nesting when you're in the middle of the file. So instead of nesting everything into `basic { ... }`, I repeat the prefix for each section. Also, I reduced the level of nesting in some places (e.g. move `control-connection` out of `connection`).

Reply:
This was my first question as it initially seems odd.  It does make it more obvious so 👍 on doing it this way.

Reply:
We're spoiled in IDEA because it recognizes HOCON and displays a breadcrumbs-style widget when the caret is on an option.
But I think users will most often look at this file in a less friendly environment, such as GitHub web or the HTML documentation.